hvac/v1/__init__.py

```diff
@@ -978,6 +986,156 @@ def auth_approle(self, role_id, secret_id=None, mount_point='approle', use_token
 
         return self.auth('/v1/auth/{0}/login'.format(mount_point), json=params, use_token=use_token)
 
+    def auth_iam(self, iam_role):
+        """
+        Authenticate to Vault per https://www.vaultproject.io/docs/auth/aws.htm.
+
+        :param iam_role: IAM role
+        :return:
+        """
+        parsed = urlparse(self._url)
+        self.token = self._authenticate_to_vault(parsed.hostname,
+                                                 parsed.port,
+                                                 iam_role,
+                                                 self._kwargs['verify'])
+        return self.token
+
+    def _authenticate_to_vault(self, vault_host, vault_port, role, verify):
+        payload = self._generate_vault_request(role, vault_host)
+
+        headers = {
+            'Content-type': 'application/json',
+            'Accept': 'text/plain'  # ,
+        }
+
+        response = requests.post('https://{}:{}/v1/auth/aws/login'.format(vault_host, vault_port),
```

Comment:
Same here

Reply:
just FYI, unless you're doing this on localhost or entirely within a controlled net, you're opening yourself up to having someone with a basic packet sniffer steal you AWS creds. Probably not a huge risk, but worth thinking about.

Reply:
My suggestion is that hvac support http since Vault does as well, but yes, it should only be used in dev environment and private network.

Reply:
Wut? HTTPS / TLS is meant to encrypt data in transit...